dashboard/src/components/HandleLink/HandleLink.tsx

```diff
@@ -0,0 +1,34 @@
+import { LinkComponentProps } from 'node_modules/@tanstack/react-router/dist/esm/link';
+import React, { PropsWithChildren, useEffect, useRef, useState } from 'react';
+import { Link } from '@tanstack/react-router';
+
+const HandleLink: React.FC<PropsWithChildren<LinkComponentProps<'a'>>> = ({
+  children,
+  ...props
+}) => {
+  const [linkTarget, setLinkTarget] = useState<string | undefined>(undefined);
+  const linkRef = useRef<HTMLAnchorElement>(null);
+
+  useEffect(() => {
+    if (linkTarget) {
+      setLinkTarget(undefined);
+      linkRef.current?.click();
+    }
+  }, [linkTarget]);
```

Comment:
useEffect is a little too much for a simple link, if we want to move to server side rendering we will need to refactor to remove it